public/js/pathway-focus-behavior.component.js

```diff
@@ -0,0 +1,84 @@
+AFRAME.registerComponent("pathway-focus", {
+    schema: {
+        animationDur: {type: "number", default: 1200},
+        initialPos: {type: "vec3", default: "1 0 11"},
+        targetPos: {type: "vec3", default: "1 0 11"},
+        defaultQuat: {default: new THREE.Quaternion(0,0,0,1)}
+    },
+
+    init : function() {
+        this.zoomIn = this.zoomIn.bind(this);
+        this.zoomOut = this.zoomOut.bind(this);
+        this.zoom = this.zoom.bind(this)
+        this.createZoomOutButton = this.createZoomOutButton.bind(this);
+        this.clickListener = this.clickListener.bind(this);
+        this.el.addEventListener("click", this.clickListener);
+    },
+
+    clickListener: function (e) {
+        const initVector = (new THREE.Vector3()).copy(this.data.initialPos)
+        const targetVector = (new THREE.Vector3()).copy(this.data.targetPos)
+        try {
+            if (!initVector.equals(targetVector)) {
+                this.createZoomOutButton();
+                this.zoomIn(); 
+            }
+        }catch(e) {
+            console.log(e)
+        }
+    },
+
+   /**
+   * activates the rig to animate towards given pathway position
+   */
+    zoomIn: function() {
+        const pos = this.data.targetPos.x + " " + this.data.targetPos.y + " " + this.data.targetPos.z;
+        this.zoom(pos, "__zoomIn");
+    },
+
+    /**
+     * animates the camera rig towards initial position prior zooming in
+     */
+    zoomOut: function() {
+        const pos = this.data.initialPos;
+        const posStr = pos.x + " " + pos.y + " " + pos.z
+        this.zoom(posStr, "__zoomOut");
+    },
+
+    zoom: function(pos, eventTag) {
+        const camera = document.getElementById("main-camera");
+        const cameraRig = document.getElementById("camera-rig");
+        const networkView = document.getElementById("parent-model").components["network-view"]
+        networkView.ResetCameraRotation()
```

Comment:
some missing semicolons here

if you're using vscode it might be easier to download the `Prettier` extension and auto-format it :D

Reply:
Thats pretty handy thanks :)! Ill check out!